File: frontend/src/api/onboarding.js
import http from './http'

export const getOnboardings = (params) => http.get('/onboardings', { params })
export const getOnboarding = (id) => http.get(`/onboardings/${id}`)
export const createOnboarding = (data) => http.post('/onboardings', data)
export const updateOnboarding = (id, data) => http.put(`/onboardings/${id}`, data)
export const deleteOnboarding = (id) => http.delete(`/onboardings/${id}`)
export const submitOnboarding = (id) => http.put(`/onboardings/${id}/submit`, {})
export const withdrawOnboarding = (id) => http.put(`/onboardings/${id}/withdraw`, {})
export const approveOnboarding = (id, data) => http.put(`/onboardings/${id}/approve`, data)
export const cancelApproveOnboarding = (id) => http.put(`/onboardings/${id}/cancel-approve`, {})

File: backend/routes/onboarding.go
package routes

import (
	"oa-system/handlers"

	"github.com/gin-gonic/gin"
)

func onboardingRoutes(rg *gin.RouterGroup) {
	rg.GET("/onboardings", handlers.GetOnboardings)
	rg.GET("/onboardings/:id", handlers.GetOnboarding)
	rg.POST("/onboardings", handlers.CreateOnboarding)
	rg.PUT("/onboardings/:id", handlers.UpdateOnboarding)
	rg.PUT("/onboardings/:id/submit", handlers.SubmitOnboarding)
	rg.PUT("/onboardings/:id/withdraw", handlers.WithdrawOnboarding)
	rg.PUT("/onboardings/:id/approve", handlers.ApproveOnboarding)
	rg.PUT("/onboardings/:id/cancel-approve", handlers.CancelApproveOnboarding)
	rg.DELETE("/onboardings/:id", handlers.DeleteOnboarding)
}
